spyder/plugins/history.py

```diff
@@ -230,6 +232,11 @@ def add_history(self, filename):
         editor.toggle_wrap_mode( self.get_option('wrap') )
 
         text, _ = encoding.read(filename)
+        linebreaks = [m.start() for m in re.finditer('\n', text)]
+        maxNline = CONF.get('historylog', 'max_entries')
+        if len(linebreak) > maxNline:

```

Comment:
This is where all the tests are failing--it should be ``linebreaks`` presumably. Easy mistake to make, but one a basic code checker or a simple test (unit test or just actually testing that the functionality worked without error) would have caught.